packages/wagmi/src/index.ts

```diff
@@ -2,19 +2,23 @@ import { IBindings, RequiredSigner } from '@lens-protocol/react-web';
 import { invariant } from '@lens-protocol/shared-kernel';
 import { providers } from 'ethers';
 import { SwitchChainNotSupportedError } from 'wagmi';
-import { fetchSigner, getProvider, switchNetwork } from 'wagmi/actions';
+import { fetchSigner, getNetwork, getProvider, switchNetwork } from 'wagmi/actions';
 
 export function bindings(): IBindings {
   return {
     getProvider: async ({ chainId }) => getProvider<providers.JsonRpcProvider>({ chainId }),
     getSigner: async ({ chainId }) => {
       if (chainId) {
-        try {
-          await switchNetwork({ chainId });
-        } catch (err) {
-          // best effort to switch network, if it fails, we just ignore it
-          if (!(err instanceof SwitchChainNotSupportedError)) {
-            throw err;
+        const { chain } = getNetwork();
+
+        if (chain?.id !== chainId) {
```

Comment:
Found myself in a corner... apparently `switchNetwork` hangs if asked to switch to the current network... not sure if it's ultimately caused by latest MetaMask.